### PR TITLE
Week 6 assessment elmer figueroa

### PR DESCRIPTION
**Pull request by drfudd2001**
# Submitting Week 6 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -43,4 +173,73 @@ const cubeAndSum1 = [2, 3, 4]
...
+// Ran all test suites.
+// ✨  Done in 1.19s.
+
+// Refactor:
+
+// This function takes an array as an argument
+const sumNumbersCubed = (array) => {
+  // Use the `reduce` method to compute the sum of the cubed numbers in the array
+  return array.reduce((sum, num) => sum + Math.pow(num, 3), 0)
+}
```
Comment (triciasykes) [inline code-challenges.test.js:245]: As expected, beautiful, elegant code, Elmer!  I appreciate your steps as pseudocode above each line of code. You have great coding skills.

File: interview-questions.md [hunk 2]
```diff
@@ -10,34 +10,121 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+    6. As a user, I want to be able to categorize tasks into different categories or tags, so that I can easily filter and organize my tasks based on their types or priorities.
+
+    7. As a user, I want to be able to set reminders or notifications for tasks with upcoming due dates, so that I can be alerted in advance and avoid missing any deadlines.
+
+    8. As a user, I want to be able to search for tasks based on their titles or descriptions, so that I can quickly find specific tasks or keywords within my to-do list.
+
+    9. As a user, I want to be able to prioritize tasks by assigning different levels of importance or urgency, so that I can focus on the most critical tasks first and manage my workload effectively.
+
+    10. As a user, I want to be able to view completed tasks separately from pending tasks, so that I can review my progress and keep track of my completed tasks for reference.
```
Comment (triciasykes) [inline interview-questions.md:130]: Amazing job here, Elmer!  Your details descriptions show a deep level of understanding of all the concepts.  Excellent work!

File: rails-commenting.rb [hunk 3]
```diff
@@ -48,15 +49,15 @@ def update
...
       redirect_to blog_posts_path
     end
   end
 
-  # ---9)
+  # ---9) The private keyword denotes that the following methods are private and can only be accessed within the BlogPostsController class.
   private
   def blog_post_params
-    # ---10)
+    # ---10) The blog_post_params method is a private method that filters and permits only the :title and :content parameters from the form submissions, which helps protect against unauthorized or malicious data being passed into the application. This method is used in the create and update actions to ensure that only the allowed parameters are used to create or update a blog post.
```
Comment (triciasykes) [inline rails-commenting.rb:60]: I'm trying to come up with some constructive criticism, but you won't let me!  Well done! ⭐️

**Review by triciasykes — APPROVED** [review]
Elmer! Congratulations on completing your final assessment! The work you provided in each of the areas was extremely high quality.  I'm super impressed by your detailed explanations and clean, efficient code.  I have the utmost confidence in you to take your skills to the next level.  Believe in yourself, Elmer - you have a lot to be proud of.  Not surprisingly, you have maintained your Competent skill level. 🎉

**Comment by drfudd2001** [comment]
I can express the way your praise makes me feel. I showed your comments to my wife and she told me that the only one that doesn't believe in me is myself. Thank you ☺️

**Pull request merged**